rate_limiter/src/lib.rs

```diff
@@ -461,24 +423,42 @@ impl RateLimiter {
     ///
     /// If the rate limiter is disabled or is not blocked, an error is returned.
     pub fn event_handler(&mut self) -> Result<(), Error> {
-        match self.timer_fd.as_mut() {
-            Some(timer_fd) => {
-                // Read the timer_fd and report error if there was no event.
-                match timer_fd.read() {
-                    0 => Err(Error::SpuriousRateLimiterEvent(
-                        "Rate limiter event handler called without a present timer",
-                    )),
-                    _ => {
-                        self.timer_active = false;
-                        Ok(())
-                    }
-                }
-            }
-            None => Err(Error::SpuriousRateLimiterEvent(
+        match self.timer_fd.read() {
+            0 => Err(Error::SpuriousRateLimiterEvent(
                 "Rate limiter event handler called without a present timer",
             )),
+            _ => {
+                self.timer_active = false;
+                Ok(())
+            }
         }
     }
+
+    /// Updates the parameters of the token buckets associated with this RateLimiter.
+    // TODO: Pls note that, right now, the buckets become full after being updated.
+    pub fn update_buckets(&mut self, bytes: Option<TokenBucket>, ops: Option<TokenBucket>) {
+        // TODO: We have to call make_bucket instead of directly assigning the bytes and/or ops
```

Comment:
is this still the case? do we still build via deser any structs with internal logic?

If not, at least this comment should change so that future work here can pick up a valid description for the TODO.

Reply:
Done.